src/bp/ui-admin/src/Pages/Workspace/CreateBotModal.jsx

```diff
@@ -0,0 +1,154 @@
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import _ from 'lodash'
+import { Button, Modal, ModalHeader, ModalBody, FormGroup, FormFeedback, Label, Input } from 'reactstrap'
+import { MdGroupAdd } from 'react-icons/lib/md'
+import Select from 'react-select'
+
+import api from '../../api'
+import { fetchBotTemplates, fetchBotCategories } from '../../reducers/bots'
+
+const defaultState = {
+  name: '',
+  template: null,
+  category: null,
+  error: null
+}
+
+class CreateBotModal extends Component {
+  state = { ...defaultState }
+
+  componentDidMount() {
+    if (!this.props.botCategoriesFetched) {
+      this.props.fetchBotCategories()
+    }
+    if (!this.props.templateFetched) {
+      this.props.fetchBotTemplates()
+    }
+  }
+
+  isFormValid = () => {
+    return this.formEl && this.formEl.checkValidity() && this.state.template
+  }
+
+  handleNameChanged = e => {
+    this.setState({ name: e.target.value })
+  }
+
+  handleTemplateChanged = template => {
+    this.setState({ template })
+  }
+
+  handleCategoryChanged = category => {
+    this.setState({ category })
+  }
+
+  stanitizeName = () => {
+    return this.state.name
+      .toLowerCase()
+      .replace(/\s/g, '-')
+      .replace(/[$&+,:;=?@#|'<>.^*()%!]/g, '')
+  }
+
+  createBot = async e => {
+    e.preventDefault()
+
+    if (!this.isFormValid()) {
+      return
+    }
+
+    const id = this.stanitizeName()
+    const name = this.state.name
+    const template = _.pick(this.state.template, ['id', 'moduleId'])
+    const category = this.state.category.value
```

Comment:
If no categories exists, it gives a null pointer

Reply:
fixed